pass/resource_pass_password.go

```diff
@@ -24,70 +22,101 @@ func passPasswordResource() *schema.Resource {
 				Type:        schema.TypeString,
 				Required:    true,
 				ForceNew:    true,
-				Description: "Full path where the pass data will be written.",
+				Description: "Full path where the pass data will be written",
 			},
 
 			"password": {
 				Type:        schema.TypeString,
-				Required:    true,
-				Description: "secret password.",
+				Optional:    true,
+				Description: "Secret password",
 				Sensitive:   true,
 			},
 
 			"data": {
 				Type:        schema.TypeMap,
 				Optional:    true,
-				Description: "additional secret data.",
+				Description: "Additional key-value data",
 				Sensitive:   true,
 			},
+
+			"yaml": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "YAML encoded data",
+				Sensitive:   true,
+			},
+
+			"body": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Body of the secret",
+			},
+
+			"full": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Entire secret contents",
+			},
 		},
 	}
 }
 
 func passPasswordResourceWrite(d *schema.ResourceData, meta interface{}) error {
 	path := d.Get("path").(string)
+	log.Printf("writing secret to path %s", path)
 
-	pp := meta.(*PassProvider)
+	pp := meta.(*passProvider)
 	pp.mutex.Lock()
 	defer pp.mutex.Unlock()
 	st := pp.store
 
-	passwd := d.Get("password").(string)
-
+	password := d.Get("password").(string)
 	data := d.Get("data").(map[string]interface{})
-	dataYaml, err := yaml.Marshal(&data)
-	if err != nil {
-		return errors.Wrapf(err, "failed to marshal data as YAML for %s", path)
+	yaml := d.Get("yaml").(string)
+
+	if len(data) != 0 && yaml != "" {
+		return errors.New("can't set data and yaml at the same time")
 	}
 
-	if len(data) == 0 {
-		sec := secret.New(passwd, fmt.Sprintf(""))
-		err = st.Set(context.Background(), path, sec)
+	value := password
+
+	if yaml != "" {
+		value += "\n---\n" + yaml
 	} else {
-		sec := secret.New(passwd, fmt.Sprintf("---\n%s", dataYaml))
-		err = st.Set(context.Background(), path, sec)
+		var result bytes.Buffer
+		tmpl, err := template.New("secret").Parse(`
+{{ range $key, $value := .Data }}
+{{ $key }}: {{ $value }}
+{{ end }}`)
+		err = tmpl.Execute(&result, &templateData{data})
+		if err != nil {
+			return errors.New("error when rendering secret template")
+		}
+		value += result.String()
 	}
```

Comment:
You could replace the template with something along the lines of:

```go
  elems := make([]string, len(data))
  for k, v := range data {
    elems = append(elems, fmt.Sprintf("%s: %s", k, v))
  }

  value := strings.Join(elems, "\n")
```

Reply:
@toabi did you see this comment?

Reply:
Yeah. Sorry I didn't have much time recently to work on this more.

Reply:
@toabi will you have some time or should re-fork your fork and redo this PR with changes?

Reply:
@jaceq You're welcome to re-fork my fork to get this freaking fix flying 🥳